js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -1440,22 +1450,23 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
     _focusInHandler: function(e) {
         this._updateFocusState(e, true);
 
-        const isSelectAllItem = $(e.target).hasClass(SELECT_ALL_ITEM_CLASS);
+        const focusedElement = this.option('focusedElement');
 
-        if(isSelectAllItem || this.option('focusedElement')) {
+        if(focusedElement) {
             clearTimeout(this._setFocusedItemTimeout);
 
-            const element = isSelectAllItem ? getPublicElement(this._$selectAllItem) : $(this.option('focusedElement'));
-
             this._setFocusedItemTimeout = setTimeout(() => {
-                this._setFocusedItem(element);
+                this._optionChanged({ name: 'focusedElement', value: focusedElement });
```

Comment:
This code looks really strange to me. If we already have `focusedElement = x`, it means that `optionChanged` was already called for this. `optionChanged` call is not very intuitive here, because no option was changed.  If we really need it, it's better call the operations directly.

I believe it should look like this:
```javascript
const $focusedElement = $(this.option('focusedElement'));
this._setFocusedItem($focusedElement);
```

Moreover, no test is failed if I remove this line. Please, add it. Seems like there should be a test "focus SelectAll -> unfocus treeView ->  focus treeView -> SelectAll is focused" and "focus second item -> unfocus treeView ->  focus treeView -> second item is focused"

Reply:
good catch, there was a test but i broke it with my last commit) i'll get it back